### PR TITLE
feat (social login): implementation of native social login

### PR DESCRIPTION
**Pull request by cerebrl**
**Summary:**

Implementation of Social Login to be handled natively, rather than through Centralized Login only. Identity Providers supported are Google and Facebook.

**Details:**

New callbacks added:

- `SelectIdPCallback`: for IDM Social Login nodes
- `RedirectCallback`: for both AM and IDM Social Login nodes

New FRAuth methods added for handling the redirection and resuming of an authentication tree:

- `FRAuth.redirect`: Saves the current step and triggers the redirection to IdP
- `FRAuth.resume`: Handles the return to the authentication journey, retrieves previous step information, if appropriate, and calls next with options

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,18 @@
 # Changelog
 
+## [Unreleased]
+
+### Added
+
+- Native Social Login callbacks for Identity Providers Google and Facebook
+- New `FRAuth` methods for handing redirection to provider and resuming an authentication journey
+  - `FRAuth.resume` supports both return from an IdP and returning from Email Suspend node
+- New `FRAuth.start` method that aliases `FRAuth.next` to align native mobile SDKs
```
Comment (cerebrl) [inline CHANGELOG.md:10]: I just added this method to align with native mobile SDKs.

File: src/fr-auth/index.ts [hunk 2]
```diff
@@ -67,9 +71,102 @@ abstract class FRAuth {
...
+  }
+
+  /**
+   * Requests the first step in the authentication tree.
+   * This is essentially an alias to calling FRAuth.next without a previous step.
+   *
+   * @param options Configuration overrides
+   * @return The next step in the authentication tree
+   */
+  public static async start(
```
Comment (cerebrl) [inline src/fr-auth/index.ts:164]: This newly added method for SDK alignment is just essentially an alias to `FRAuth.next` in order to align with native mobile.

File: tests/e2e/suites/authn-social-login-am.test.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * Copyright (c) 2020 ForgeRock. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+import { setupAndGo } from '../utilities/setup-and-go';
+import browsers from '../utilities/browsers';
+
+describe('Test bad login flow', () => {
```
Comment (spetrov) [inline tests/e2e/suites/authn-social-login-am.test.ts:14]: This looks like a successful login flow?

File: tests/e2e/suites/authn-social-login-am.test.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  browsers.forEach((browserType) => {
+    it(`Login with device profile callback ${browserType}`, async (done) => {
```
Comment (spetrov) [inline tests/e2e/suites/authn-social-login-am.test.ts:16]: Social login with AM nodes?

File: tests/e2e/suites/authn-social-login-idm.test.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * Copyright (c) 2020 ForgeRock. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+import { setupAndGo } from '../utilities/setup-and-go';
+import browsers from '../utilities/browsers';
+
+describe('Test bad login flow', () => {
```
Comment (spetrov) [inline tests/e2e/suites/authn-social-login-idm.test.ts:14]: Same here...

File: tests/e2e/suites/authn-social-login-idm.test.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  browsers.forEach((browserType) => {
+    it(`Login with device profile callback ${browserType}`, async (done) => {
```
Comment (spetrov) [inline tests/e2e/suites/authn-social-login-idm.test.ts:16]: Successful social login with IDM nodes?

**Review by spetrov — CHANGES_REQUESTED** [review]
I tested manually social login with google and facebook with trees which use AM and IDM nodes and did some error case testing (mostly focusing on the FRAuth.redirect(step) and FRAuth.resume(url, options) methods to ensure that step information is properly stored and extracted from local storage (and then removed from there...)

There are 2 minor "test naming" mistakes that I've noticed. 
Other than that everything looks good to me! 👍

**Comment by cerebrl** [comment]
@spetrov PR has been updated from your feedback. Thanks for catching the test naming.

**Review by spetrov — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**